src/System.Data.SqlClient/src/System/Data/SqlClient/SqlClientDiagnosticListenerExtensions.cs

```diff
@@ -205,5 +217,182 @@ public static void WriteConnectionCloseError(this DiagnosticListener @this, Guid
+                        Connection = connection,
+                        Timestamp = Stopwatch.GetTimestamp()
+                    });
+
+                return operationId;
+            }
+            else
+                return Guid.Empty;
+        }
+
+        public static void WriteTransactionCommitAfter(this DiagnosticListener @this, Guid operationId, IsolationLevel isolationLevel, SqlConnection connection, [CallerMemberName] string operation = "")
+        {
+            if (@this.IsEnabled(SqlAfterCommitTransaction))
+            {
+                @this.Write(
+                    SqlAfterCommitTransaction,
+                    new
+                    {
+                        OperationId = operationId,
+                        Operation = operation,
+                        IsolationLevel = isolationLevel,
+                        Connection = connection,
+                        Timestamp = Stopwatch.GetTimestamp()
+                    });
+            }
+        }
+
+        public static void WriteTransactionCommitError(this DiagnosticListener @this, Guid operationId, IsolationLevel isolationLevel, SqlConnection connection, Exception ex, [CallerMemberName] string operation = "")
+        {
+            if (@this.IsEnabled(SqlErrorCommitTransaction))
+            {
+                @this.Write(
+                    SqlErrorCommitTransaction,
+                    new
+                    {
+                        OperationId = operationId,
+                        Operation = operation,
+                        IsolationLevel = isolationLevel,
+                        Connection = connection,
+                        Exception = ex,
+                        Timestamp = Stopwatch.GetTimestamp()
+                    });
+            }
+        }
+
+        public static Guid WriteTransactionRollbackBefore(this DiagnosticListener @this, IsolationLevel isolationLevel, SqlConnection connection, string transactionName, [CallerMemberName] string operation = "")
+        {
+            if (@this.IsEnabled(SqlBeforeRollbackTransaction))
+            {
+                Guid operationId = Guid.NewGuid();
+
+                @this.Write(
+                    SqlBeforeRollbackTransaction,
+                    new
+                    {
+                        OperationId = operationId,
+                        Operation = operation,
+                        IsolationLevel = isolationLevel,
+                        Connection = connection,
+                        TransactionName = transactionName,
+                        Timestamp = Stopwatch.GetTimestamp()
+                    });
+
+                return operationId;
+            }
+            else
+                return Guid.Empty;
+        }
+
+        public static void WriteTransactionRollbackAfter(this DiagnosticListener @this, Guid operationId, IsolationLevel isolationLevel, SqlConnection connection, string transactionName, [CallerMemberName] string operation = "")
```

Comment:
@saurabh500 Same as comment above, see the convention in the file: WriteConnectionOpenAfter.